synapse/replication/tcp/redis/factory.py

```diff
@@ -0,0 +1,46 @@
+from synapse.replication.tcp.redis.connection import IRedisConnection, SentinelRedisConnection, RedisConnection
+from synapse.replication.tcp.redis.replication import RedisReplicationFactory, RedisSentinelReplicationFactory
+import txredisapi
+from typing import List, TYPE_CHECKING
+if TYPE_CHECKING:
+    from synapse.server import HomeServer
+
+
+def get_redis_connection(hs: "HomeServer", config) -> "IRedisConnection":
+    if config.sentinel_enabled:
+        return SentinelRedisConnection(
+            hs=hs,
+            sentinels=config.sentinels,
+            service_name=config.service_name,
+            password=config.redis_password,
+            reconnect=True)
+    else:
+        return RedisConnection(
+            hs=hs,
+            host=config.redis_host,
+            port=config.redis_port,
+            password=config.redis_password,
+            reconnect=True)
+
+
+def get_replication_factory(hs: "HomeServer", connection: IRedisConnection, channel_names: List[str]) -> txredisapi.RedisFactory:
```

Comment:
The multi line method params is used multiple times. Maybe we should apply the same approach to methods which have "long" definition?